metrics/ci_integrations/test/integration/ci/sync_stage/builds/new_builds_sync_stage_test.dart

```diff
@@ -0,0 +1,495 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:ci_integration/integration/ci/config/model/sync_config.dart';
+import 'package:ci_integration/integration/ci/sync_stage/builds/new_builds_sync_stage.dart';
+import 'package:metrics_core/metrics_core.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../../../cli/test_util/mock/integration_client_mock.dart';
+import '../../../../cli/test_util/test_data/config_test_data.dart';
+import '../../../../test_utils/matchers.dart';
+
+void main() {
+  group("NewBuildsSyncStage", () {
+    const build = BuildData(buildNumber: 1);
+    const anotherBuild = BuildData(buildNumber: 2);
+    const builds = [build, anotherBuild];
+
+    final coverage = Percent(0.1);
+    final buildWithCoverage = build.copyWith(coverage: coverage);
+    final anotherBuildWithCoverage = anotherBuild.copyWith(coverage: coverage);
+    final buildsWithCoverage = [buildWithCoverage, anotherBuildWithCoverage];
+
+    final sourceClient = SourceClientMock();
+    final destinationClient = DestinationClientMock();
+
+    final newBuildsSyncStage = NewBuildsSyncStage(
+      sourceClient,
+      destinationClient,
+    );
+
+    final syncConfig = ConfigTestData.syncConfig;
+    final sourceProjectId = syncConfig.sourceProjectId;
+    final destinationProjectId = syncConfig.destinationProjectId;
+    final initialSyncLimit = syncConfig.initialSyncLimit;
+    final syncConfigWithCoverage = SyncConfig(
+      sourceProjectId: sourceProjectId,
+      destinationProjectId: destinationProjectId,
+      initialSyncLimit: initialSyncLimit,
+      inProgressTimeout: syncConfig.inProgressTimeout,
+      coverage: true,
+    );
```

Comment:
It seems like the `ConfigTestData` is strongly related to config parsing for more complex tests. So maybe here we can make it clearer: 
```suggestion
    final syncConfig = SyncConfig(
      // test data goes here
    );
```
And where necessary to match the arguments: 
```dart
  final expectedId = syncConfig.destinationProjectId;
```